File: secante/secante.jl
#Newton
#La funcion a evaluar es f(x)=0.5p^(-0.2)+0.5p^(-0.5)-2=0
#Parametros
diff=1
tol=1*10.0^(-6)
contador=0

#Funcion
f = (x) -> 0.5*float(x)^(-0.2)+0.5*float(x)^(-0.5)-2

#Puntos Iniciales
x0=0.1
x_1=0.3

#Busqueda
while diff > tol
  fx0=f(x0)
  fx_1=f(x_1)
  cociente=(x0-x_1)/(fx0-fx_1)
  x=x0-(cociente*fx0)
  diff=abs(x-x0)
  x_1=x0
  x0=x
  contador=contador+1
@printf "Iteracion %d, diferencia %9.7f\n" contador diff
if contador > 200
    println("Error: Maximo numero de iteraciones alcanzado")
    break
end
end
fx=f(x)
@printf "La raiz es %9.7f y f(x) %9.9f" x fx

File: secante/secante.f90
! PROGRAM secante

program newton

    ! variable declaration
    implicit none
    integer :: contador
    real*8 :: x, xo, x_1, fx, diff, tol, fxo, fx_1, cociente
    

    ! Parametros
    diff=1
    tol=1d-6
    contador=0

    ! Puntos iniciales
    xo=0.1d0
    x_1=0.3d0
    

    ! Busqueda
    do while (diff > tol)

      fxo=testfun(xo)
      fx_1=testfun(x_1)
      cociente=(xo-x_1)/(fxo-fx_1)
      x=xo-cociente*fxo
      diff=abs(x-xo)
      x_1=xo
      xo=x
      contador=contador +1

      if (contador > 200)then
        write(*,'(/a)')' Numero maximo de iteraciones alcanzado '
        stop
      endif
      

      write(*,'(/a,i4,a,f12.7)')' Iteracion  ',contador,'  diferencia ',diff,
      
    enddo

    fx=testfun(x)

    write(*,'(/a,f12.7,a,f12.9)')' x = ',x,'    f = ',fx,


    contains

    function testfun(x)

        implicit none
    
        ! input arguments
        real*8, intent(in) :: x
    
        ! function value
        real*8 :: testfun
          
        ! executable code
        testfun = 0.5d0*x**(-0.2d0)+0.5d0*x**(-0.5d0)-2d0
    
    end function


end program